### ISSUE TITLE
Allow realm DynamicLibrary to be loaded from a different path than the current directory

### ISSUE THREAD

**Issue by nsivraj** [realm/realm-dart#877]
### Description

1) Goals - Allow realm DynamicLibrary to be loaded from a different path than the current directory
2) Expected results - When running a dart compiled exe allow the realm DynamicLibrary to be loaded from a different path than the current directory
3) Actual results - When running a dart compiled exe from the command line the librealm_dart.dylib library cannot be found
4) Steps to reproduce - Try to run a dart compiled exe which uses the realm DynamicLibrary from a directory other than the one it was compiled in and it fails with the error: Invalid argument(s): Failed to load dynamic library '/Users/normanjarvis/forge/personal/code/coderockit/cr-e2e-tests/binary/macos/librealm_dart.dylib': dlopen(/Users/normanjarvis/forge/personal/code/coderockit/cr-e2e-tests/binary/macos/librealm_dart.dylib, 0x0001): tried: '/Users/normanjarvis/forge/personal/code/coderockit/cr-e2e-tests/binary/macos/librealm_dart.dylib' (no such file)
5) Code sample that highlights the issue (full Flutter or Dart project that we can run ourselves are ideal) - Please see the code here in this github repo https://gitlab.com/coderockit/cr-cli
6) Version of Realm (0.4.0+beta), Flutter(3.0.3) and Dart (2.17.5)
7) Version of desktop OS - Mac, Windows or Linux - Mac macOS Monterey Version 12.4 Apple M1 Max
8) Version of target mobile OS - this is a command line tool for a command prompt

### How important is this improvement for you?

Dealbreaker

**Comment by blagoev** [comment]
Hi, We would like to understand the problem more. 
Right now if you compile a dart executable and you copy it to a different directory than the one it was compiled from, the Dart app will continue to work but as the error says, you need to copy the `binary` directory containing the realm binary as well next to the executable. 
So you will have this folder structure
```
/Users/normanjarvis/forge/personal/code/coderockit/cr-e2e-tests/myapp   <--- the dart executable
/Users/normanjarvis/forge/personal/code/coderockit/cr-e2e-tests/binary/macos/librealm_dart.dylib <---- the realm binary
```
We would like to understand the reasons why you can not do that. Is that not working for you?

**Comment by nielsenko** [comment]
We grok the problem. See comments on: #877 and #881

**Comment by nielsenko** [comment]
@nsivraj Fix released with [0.5.0+beta](https://pub.dev/packages/realm/versions/0.5.0+beta)

**Comment by nsivraj** [comment]
@nielsenko Thank you very much. I will try release 0.5.0+beta.
